buffer/src/main/java/io/netty/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -166,15 +166,15 @@ public boolean readOnly() {
     public Buffer copy(int offset, int length) {
         checkLength(length);
         checkGet(offset, length);
-        AllocatorControl.UntetheredMemory memory = control.allocateUntethered(this, length);
-        ByteBuffer base = memory.memory();
-        ByteBuffer buffer = length == 0? bbslice(base, 0, 0) : base;
-        Drop<NioBuffer> drop = memory.drop();
-        NioBuffer copy = new NioBuffer(base, buffer, control, drop);
-        drop.attach(copy);
-        copyInto(offset, copy, 0, length);
-        copy.writerOffset(length);
-        return copy;
+        Buffer copy = control.getAllocator().allocate(length);
+        try {
+            copyInto(offset, copy, 0, length);
+            copy.writerOffset(length);
+            return copy;
+        } catch (Exception e) {
```

Comment:
Let's catch `Throwable` just in case ...

```suggestion
        } catch (Throwable e) {
```